Frontend/src/app/query-interface/properties/page.tsx

```diff
@@ -0,0 +1,175 @@
+import { ISelectProp } from "@/types/user-types";
+import type { FormProps } from 'antd';
+import { analyzeGraph } from "@/services/analyzer-service";
+
+// Render input field based on InputTypes
+const renderInputField = (key: string, inputParam: InputParam) => {
+  switch (inputParam.type) {
+    case InputTypes.INPUT:
+      return (
+        <Form.Item label={inputParam.label || key} name={key}>
+          <Input />
+        </Form.Item>
+      );
+    case InputTypes.INPUTNUMBER:
+      return (
+        <Form.Item label={inputParam.label || key} name={key}>
+          <InputNumber />
+        </Form.Item>
+      );
+    case InputTypes.SWITCH:
+      return (
+        <Form.Item label={inputParam.label || key} name={key} valuePropName="checked">
+          <Switch />
+        </Form.Item>
+      );
+    case InputTypes.SELECT:
+      return (
+        <Form.Item label={inputParam.label || key} name={key}>
+          <Select>
+            {inputParam.options?.map((option, index) => (
+              <Select.Option key={index} value={option.value}>
+                {option.label}
+              </Select.Option>
+            ))}
+          </Select>
+        </Form.Item>
+      );
+    default:
+      return null;
+  }
+};
+
+export default function QueryPropoerties() {
+  const { token } = theme.useToken();
+
+  const [graphs, setGraphs] = useState<ISelectProp[]>([]);
+  const [isAnalysing, setIsAnalysing] = useState<boolean>(false);
+  const [selectedTool, setSelectedTool] = useState<GraphAnalyzer>();
+
+  const getGraphsData = async () => {
+    try{
+    const res = await getGraphList();
+    if(res.data){
+      const filteredData: ISelectProp[] = res.data.map((graph: any) => {
+         return {
+          value: graph.idgraph,
+          label: graph.name
+        }
+      })
+      setGraphs(filteredData);
+    }
+    }catch(err){
+      message.error("Failed to fetch the list of graphs");
+    }
+  }
+
+  useEffect(() => {
+    getGraphsData();
+  }, [])
+
+  const listStyle: React.CSSProperties = {
+    lineHeight: '200px',
+    textAlign: 'center',
+    background: token.colorFillAlter,
+    borderRadius: token.borderRadiusLG,
+    marginTop: 16,
+  };
+  
+  type FieldType = {
+    graph_id?: string;
+    method?: string;
+  };
+
+  const runAnalysis = async (values: any) => {
+    const result = await analyzeGraph(values);
+    if (result.data) {
+      message.success("Analysis completed successfully");
+    } else {
+      message.error("Error in graph analysis: " + result.message);
+    }
+    setIsAnalysing(false);
+  }
+  
+  const onValueChange = (param: any) => {
+    if (param && param.method) {
+      const tool = Analyzers.find((tool) => tool.name == param.method || tool.id == param.method)
+      setSelectedTool(tool);
+    }
+  }
+
+  const onFinish: FormProps<FieldType>['onFinish'] = async (values) => {
+    setIsAnalysing(true);
+    await runAnalysis(values);
+  };
+
+  return (
+    <>
```

Comment:
Why do we need `<>` ?

Reply:
In React, we can only return a single component. That's why I wrap all the components inside a fragment (<> </>) before returning them.

Reply:
Ok